src/Node.java

```diff
@@ -0,0 +1,42 @@
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+
+public class Node {
+
+    private boolean endOfWord;
+    private final char value;
+    private ArrayList<Node> edges = new ArrayList<>();
+
+    Node(char value) {
+        this.value = value;
+    }
+
+    public boolean isEndOfWord() {
+        return endOfWord;
+    }
+
+    public void setEndOfWord(boolean endOfWord) {
+        this.endOfWord = endOfWord;
+    }
+
+    public char getValue() {
+        return value;
+    }
+
+    public Collection<Node> getEdges() {
+        return Collections.unmodifiableCollection(edges);
+    }
```

Comment:
Thinking of reducing this even further, we don't need a collection here really. All we care about is finding the edge for a particular character. It makes sense to move the search here (the lambda) and transform this into a method that returns an `Optional<Node>` for a given character.